joint_state_publisher/CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ install(PROGRAMS joint_state_publisher/joint_state_publisher DESTINATION ${CATKI
 
 
 if(CATKIN_ENABLE_TESTING)
-  find_package(catkin REQUIRED COMPONENTS rostest)
+  find_package(rostest)
```

Comment:
You shouldn't re-find `catkin`. Just find the additional package you need separately.